arch/risc-v/src/mpfs/mpfs_corespi.c

```diff
@@ -720,6 +731,52 @@ static int mpfs_spi_sem_waitdone(struct mpfs_spi_priv_s *priv)
   return nxsem_tickwait_uninterruptible(&priv->sem_isr, USEC2TICK(timeout));
 }
 
+/****************************************************************************
+ * Name: mpfs_spi_write_tx8
+ *
+ * Description:
+ *   Fill up the TX fifo with one word in configured size frames
+ *
+ * Input Parameters:
+ *   priv     - SPI private state data
+ *   nframes  - Number of 8-bit data per word
+ *   last     - true if this is the last word of the transfer
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static inline void mpfs_spi_write_tx8(struct mpfs_spi_priv_s *priv,
+                                      unsigned nframes,
+                                      bool last)
+{
+  uint8_t *data8 = (uint8_t *)priv->txbuf;
+  unsigned pos = priv->tx_pos++;
+
+  /* Calculate octet position in the tx-buffer */
+
+  pos *= nframes;
+
+  /* Write all but last 8-bit data */
+
+  while (--nframes > 0)
+    {
+      putreg32((uint32_t)data8[pos + nframes], MPFS_SPI_TX_DATA);
```

Comment:
Writes the data in reverse, or am I missing something ?

Reply:
Yes, the SPI frame is sent msb bit first, so making a 16-bit transfer out of 2 8-bit transfers requires sending the msb byte first, lsb byte last.